src/ng/http.js

```diff
@@ -141,6 +141,9 @@ function defaultHttpResponseTransform(data, headers) {
         try {
           data = fromJson(tempData);
         } catch (e) {
+          if (!contentType || contentType && contentType.indexOf(APPLICATION_JSON)) {
```

Comment:
The `contentType && ` part is unnecessary (since we are already on the right-hand side of the `||`, where `contentType` is truthy. So, the above coondition would become `!contentType || contentType.indexOf(APPLICATION_JSON)`.

While the expression works as intended, it is a little cryptic/hard to read. Since you are essentially trying to negate the already checked `contentType && (contentType.indexOf(APPLICATION_JSON) === 0)`, I suggest you refactor it to something like:

```diff
 if (tempData) {
   var contentType = headers('Content-Type');
-  if ((contentType && (contentType.indexOf(APPLICATION_JSON) === 0)) || isJsonLike(tempData)) {
+  var hasJsonContentType = contentType && (contentType.indexOf(APPLICATION_JSON) === 0);
+  if (hasJsonContentType || isJsonLike(tempData)) {
     try {
       data = fromJson(tempData);
     } catch (e) {
+      if (!hasJsonContentType) {
+        return data;
+      }
       throw $httpMinErr(...);
     }
```